### PR TITLE
CSF201 registration form

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adding the new questions to the CSF201 workshop enrollment form and showing those answers in the workshop dashboard. This does not turn on the ability to create a CSF201 workshop.

Workshop dashboard:
screenshot 2019-01-29 at 3 41 12 pm

Enrollment form:
screenshot 2019-01-30 at 4 20 21 pm
screenshot 2019-01-30 at 4 20 34 pm

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx [hunk 1]
```diff
@@ -309,6 +379,51 @@ export default class EnrollForm extends React.Component {
...
+            onChange={this.handleChange}
+            selectedItems={this.state.csf_courses_planned}
+            validationState={this.state.errors.hasOwnProperty("csf_courses_planned") ? VALIDATION_STATE_ERROR : null}
+            errorText={this.state.errors.csf_courses_planned}
+            type="check"
+          />
+          <ButtonList
+            id="attended_csf_intro_workshop"
+            key="attended_csf_intro_workshop"
+            answers={Object.keys(ATTENDED_CSF_COURSES_OPTIONS).map( key => key)}
```
Comment (clareconstantine) [inline apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx]: You don't need to do the `map` here - `Object.keys(ATTENDED_CSF_COURSES_OPTIONS)` returns the same thing

**Review by clareconstantine — APPROVED** [review]
One tiny comment but overall this looks good!

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx [hunk 2]
```diff
@@ -113,6 +147,25 @@ export default class EnrollForm extends React.Component {
     return roleWithDescription;
   }
 
+  csfCoursesPlanned() {
+    if (!this.state.csf_courses_planned) {
+      return undefined;
+    }
+    const processedCourses = [];
+    this.state.csf_courses_planned.forEach((g) => {
+      if (g === `${OTHER} ${EXPLAIN}`) {
```
Comment (hacodeorg) [inline apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx:156]: What is this magic `===` ?

Comment (bethanyaconnor) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx:156]: The triple equals checks that both value and type are the same. Potentially overkill in this particular block but it's consistent with the surrounding code and is safer anyway.

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment_school_info.jsx [hunk 3]
```diff
@@ -45,6 +46,16 @@ export default class WorkshopEnrollmentSchoolInfo extends React.Component {
     this.props.onDelete(pendingDeleteId);
   }
 
+  csfCourseExperience(csf_course_experience) {
```
Comment (hacodeorg) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment_school_info.jsx]: I'm curious about our naming convention for method, should it start with a verb, or we consider this as an attribute?

**Review by hacodeorg — APPROVED** [review]

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx]: Thanks! Fixed.

File: apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx (hunk 2 above)
Comment (clareconstantine) [reply apps/src/code-studio/pd/workshop_enrollment/enroll_form.jsx:156]: Our linter requires it in many scenarios, although I'm not sure exactly what the rule is

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment_school_info.jsx (hunk 3 above)
Comment (bethanyaconnor) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_enrollment_school_info.jsx]: I think it should start with a verb. I'll fix that.

**Pull request merged**